core/main/src/state/cap/generic_cap_state.rs

```diff
@@ -120,4 +122,34 @@ impl GenericCapState {
         self.check_supported(permissions)?;
         self.check_available(permissions)
     }
+
+    pub fn clear_non_negotiable_permission(
```

Comment:
Can you confirm if this function removes non-negotiable permission from the input permissions list ?